added mpa-7103 - basic CRUD tests for configuration service

PavloEsco opened the pull request:
Automation testing for the configuration service - basic CRUD tests

PavloEsco committed:
added mpa-7103 - basic CRUD tests for configuration service

PavloEsco committed:
added basic tests for configuration service using the instrument instance

PavloEsco committed:
added basic tests for configuration service using the instrument instance

PavloEsco committed:
Merge branch 'develop' into develop-mpa-7103-Automation-testing-for-the-configuration-service-basic-CRUD-TC-s

PavloEsco committed:
Merge remote-tracking branch 'origin/develop-mpa-7103-Automation-testing-for-the-configuration-service-basic-CRUD-TC-s' into develop-mpa-7103-Automation-testing-for-the-configuration-service-basic-Instance-TC-s

# Conflicts:
#	scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature

PavloEsco committed:
added test for instrument instance duplicate

PavloEsco committed:
Merge branch 'develop' into develop-mpa-7103-Automation-testing-for-the-configuration-service-basic-CRUD-TC-s

aharonha commented on scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature:
```diff
@@ -0,0 +1,301 @@
+
+
+  #MPA-7103 (#2.2)
+  Scenario: Fail to save (edit) the write permission for specific entity which already exist (duplicate)
```
```suggestion
  Scenario: Update (override - edit) the value for specific entity which already exist
```

aharonha commented on scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature:
```diff
@@ -0,0 +1,301 @@
+    And "repository" with "specified" name already created and stored in DB
+    When this user requested to save the same entity (key or value) which already exists with write permission for the relevant "repository" specified name
+    Then this new entity (key or value) shouldn't be created
+    And the user should get one of the error messages: "key: "specified" key name already exists" or "value": "specified" value name already exists"
```
```suggestion
        Then a new value should be set to the given key and stored in the relevant "repository" of the DB
```

aharonha commented on scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature:
```diff
@@ -0,0 +1,301 @@
+    Given a user have got a valid "token" (API key) with assigned "owner" or "admin" roles
+    And "repository" with "specified" name already created and stored in DB
+    When this user requested to save the same entity (key or value) which already exists with write permission for the relevant "repository" specified name
+    Then this new entity (key or value) shouldn't be created
```
```suggestion
```

aharonha commented on scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature:
```diff
@@ -0,0 +1,301 @@
+
+
+  #MPA-7103 (#5.2)
+  Scenario: Fail to save a duplicate of existent instrument instance entity
```
should not fail

PavloEsco replied:
Agreed

aharonha reviewed (changes requested):
do some changes

aharonha committed:
Update scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature

Co-Authored-By: PavloPetrina <36890403+PavloPetrina@users.noreply.github.com>

aharonha committed:
Update scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature

Co-Authored-By: PavloPetrina <36890403+PavloPetrina@users.noreply.github.com>

aharonha committed:
Update scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature

Co-Authored-By: PavloPetrina <36890403+PavloPetrina@users.noreply.github.com>

PavloEsco committed:
added test for instrument instance duplicate

PavloEsco committed:
Merge remote-tracking branch 'origin/develop-mpa-7103-Automation-testing-for-the-configuration-service-basic-CRUD-TC-s' into develop-mpa-7103-Automation-testing-for-the-configuration-service-basic-Instance-TC-s

# Conflicts:
#	scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature

aharonha committed:
Update API-DOC-Configuration.md

PavloEsco committed:
Update mpa-7103-Automation-testing-for-the-configuration-servicen.feature

Added additional scenarios for save/update with symbols and delete of non-existent values

PavloEsco committed:
added API calls Json for senbox

PavloEsco committed:
added first part of new API doc as for Exchange

PavloEsco committed:
Finished API doc

PavloEsco committed:
removed API docs for separate pull

PavloEsco committed:
removed API docs

aharonha commented on scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature:
```diff
@@ -62,42 +62,41 @@ Feature: Basic CRUD tests for configuration service.
-    When this user requested to save the "specific" entity (key and value) with write permission for the relevant "repository" specified name
-    Then this new entity (key and value) shouldn't be created and the user should get an error message: "role: permission denied"
+  Scenario: Successful update (override - edit) the value for specific entity which already exist
+    Given a user have got a valid "tokens" (API key) with assigned "admin" and "owner" roles
```
token

aharonha replied:
```suggestion
     Given a user have got a valid "token" (API key) with assigned "owner" role
```

aharonha commented on scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature:
```diff
@@ -62,42 +62,41 @@ Feature: Basic CRUD tests for configuration service.
+  Scenario: Successful update (override - edit) the value for specific entity which already exist
+    Given a user have got a valid "tokens" (API key) with assigned "admin" and "owner" roles
+    And relevant "repositories" with "specified" names "Repo1" and "Repo2" already created and stored in DB with some "specific" entities
+    When this user requested to save (update) the these "specific" entities (key or value) in the relevant "repositories" name "Repo1" and "Repo2" applying the "admin" and "owner" tokens
```
```suggestion
     When this user requested to update a "specific" value in the relevant "repositories" name "Repo1" and "Repo2" applying the "admin" and "owner" tokens
```

aharonha commented on scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature:
```diff
@@ -195,38 +206,54 @@ Feature: Basic CRUD tests for configuration service.
       | instrumentId | name   | DecimalPrecision | Rounding | key                        |
       | XAG          | Silver | 4                | down     | KEY-FOR-PRECIOUS-METAL-123 |
-    Then this new entity shouldn't be created and the user should get an error message like: "instrumentInstance" "instrumentId =  XAG already exists"
+    Then existent entity values shouldn't be duplicated thus new values should be set to the given keys and stored in the relevant "repositories" of the DB
```
`existent` -> `the existing`

aharonha committed:
Merge branch 'develop' into develop-mpa-7103-Automation-testing-for-the-configuration-service-basic-CRUD-TC-s
